### PR TITLE
Populate top-level display data in yaml main.

### PR DESCRIPTION
**Pull request by robertwb**
This will be useful in case the UI wants to display it. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @polber

File: sdks/python/apache_beam/yaml/main.py [hunk 1]
```diff
@@ -51,19 +51,22 @@ def _pipeline_spec_from_args(known_args):
...
-  pipeline_spec = _pipeline_spec_from_args(known_args)
+  pipeline_yaml = _pipeline_spec_from_args(known_args)
+  pipeline_spec = yaml.load(pipeline_yaml, Loader=yaml_transform.SafeLineLoader)
 
-  with beam.Pipeline(options=beam.options.pipeline_options.PipelineOptions(
-      pipeline_args,
-      pickle_library='cloudpickle',
-      **yaml_transform.SafeLineLoader.strip_metadata(pipeline_spec.get(
-          'options', {})))) as p:
+  with beam.Pipeline(  #
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/main.py:63]: erroneous '#'

**Review by Polber — APPROVED** [review]

File: sdks/python/apache_beam/yaml/main.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/yaml/main.py:63]: This is actually to prevent some odd yapf formatting. Clarifying.

**Comment by AnandInguva** [comment]
I think this PR is causing docs test to fail -https://github.com/apache/beam/actions/runs/6461453740/job/17541145482?pr=28902

**Comment by robertwb** [comment]
Huh. codecov was the only failure in presubmits, but it looks like a bunch didn't run. Reverting for now to diagnose.

**Pull request merged**